torch/xpu/__init__.py

```diff
@@ -225,18 +226,142 @@ def _get_device(device: Union[int, str, torch.device]) -> torch.device:
     return device
 
 
+class StreamContext:
+    r"""Context-manager that selects a given stream.
+
+    All XPU kernels queued within its context will be enqueued on a selected
+    stream.
+
+    Args:
+        Stream (Stream): selected stream. This manager is a no-op if it's
+            ``None``.
+    .. note:: Streams are per-device.
+    """
+    cur_stream: Optional["torch.xpu.Stream"]
+
+    def __init__(self, stream: Optional["torch.xpu.Stream"]):
+        assert not torch.jit.is_scripting()
```

Comment:
Can you explain this assertion?

Reply:
`StreamContext` is not supported in jit script. removed this line and will raise an error in `_get_device_index` when it is jit scripting.

Reply:
Can we align the way how CUDA handles this? https://github.com/pytorch/pytorch/blob/fc5fda14bcc41008e5610bf39c53eb066933ea4e/torch/cuda/__init__.py#L499